packages/create-plugin/templates/backend/pkg/plugin/datasource.go

```diff
@@ -94,16 +94,23 @@ func (d *Datasource) query(_ context.Context, pCtx backend.PluginContext, query
 // datasource configuration page which allows users to verify that
 // a datasource is working as expected.
 func (d *Datasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
-	var status = backend.HealthStatusOk
-	var message = "Data source is working"
+	res := &backend.CheckHealthResult{}
+	config, err := models.LoadPluginSettings(*req.PluginContext.DataSourceInstanceSettings)
 
-	if rand.Int()%2 == 0 {
-		status = backend.HealthStatusError
-		message = "randomized error"
```

Comment:
Replacing this randomized error with something that is consistent so that e2e tests can be added for the health check.